sympy/polys/numberfields/subfield.py

```diff
@@ -88,45 +88,47 @@ def field_isomorphism_pslq(a, b):
     g = b.minpoly.replace(f.gen)
 
     n, m, prev = 100, b.minpoly.degree(), None
+    ctx = MPContext()
```

Comment:
Local context is a good idea but it would probably be better to use `PythonMPContext`. There are unresolved issues in the implementation of `MPContext` (see #19153). There is a single static `RealElement` (and `ComplexElement`) class whose `context` attribute is patched to point to the new `MPContext` class each time one is created. So all computations in existing `RR` and `CC` classes will always have the precision set in the last `MPContext`.

Reply:
Is this a problem with the `MPContext` imported directly from `mpmath`, or just with the one defined [in SymPy](https://github.com/sympy/sympy/blob/d822fcba181155b85ff2b29fe525adbafb22b448/sympy/polys/domains/mpelements.py#L45)?

I tried using `PythonMPContext`, but it has no `.pslq` attribute.

Reply:
In mpmath it's added through the IdentificationMethods mixin class:
https://github.com/fredrik-johansson/mpmath/blob/c11db84b3237bd8fc6721f5a0c5d7c0c98a24dc1/mpmath/identification.py#L837
That class is then a superclass for StandardBaseContext. Finally MPContext subclasses both BaseMPContext and StandardBaseContext:
https://github.com/fredrik-johansson/mpmath/blob/c11db84b3237bd8fc6721f5a0c5d7c0c98a24dc1/mpmath/ctx_mp.py#L58
No idea why it's all done in such a complicated way.

Reply:
I think we might be okay. Calling `field_isomorphism_pslq()` does not appear to be altering the context of a `RealElement`:

```python
In [2]: from sympy import *
   ...: from sympy.polys.numberfields.subfield import field_isomorphism_pslq
   ...: a = RR(2)
   ...: print(a.context.prec)
   ...: b = AlgebraicNumber(sqrt(2))
   ...: c = AlgebraicNumber(sqrt(2) + sqrt(3))
   ...: print(field_isomorphism_pslq(b, c))
   ...: print(a.context.prec)
53
[1/2, 0, -9/2, 0]
53
```